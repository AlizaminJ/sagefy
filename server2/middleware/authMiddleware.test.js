const authMiddleware = require('./authMiddleware')

describe('#authMiddleware', () => {
  test('should allow if session user', () => {
    const next = jest.fn()
    authMiddleware({ session: { user: 'abcd1234' } }, {}, next)
    expect(next).toMatchSnapshot()
  })

  test('should fail if not session user', () => {
    const next = jest.fn()
    try {
      authMiddleware({ session: {} }, {}, next)
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
    expect(next).toMatchSnapshot()
  })
})
