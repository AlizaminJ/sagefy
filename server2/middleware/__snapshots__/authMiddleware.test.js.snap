// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#authMiddleware should allow if session user 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`#authMiddleware should fail if not session user 1`] = `[Error: GlqI6VAROUCGqZy3OuKJ1A]`;

exports[`#authMiddleware should fail if not session user 2`] = `[MockFunction]`;
